bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVMScheduler.java

```diff
@@ -233,6 +235,7 @@ public void run() {
             } finally {
                 strandCountDown();
             }
+            ObserveUtils.stopCallableObservation(strand);
```

Comment:
move this one line down. and add comment saying why it is need to do this before popping the stack